### PR TITLE
Added validation policy

### LINKED ISSUE #222: Add ValidationPolicy in IngestionProperties
Add ValidationPolicy options in IngestionProperties client, that indicates which validations to run/stop during ingestion (ValidationOptions, ValidationImplications) and what to do when that is violated.

### PR DESCRIPTION
**Pull request by AsafMah**
Added validation policy to ingestion properties

Fixes #222

### CODE REVIEW

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import java.io.Serializable;
+
+public class ValidationPolicy implements Serializable {
+    private ValidationOptions validationOptions = ValidationOptions.DoNotValidate;
+    private ValidationImplications validationPolicyType = ValidationImplications.BestEffort;
+
+    public ValidationPolicy() {
+    }
+
+    public ValidationPolicy(ValidationOptions validationOptions, ValidationImplications validationPolicyType) {
```
Comment (yihezkel) [inline ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java:12]: What if they want to provide just 1 or the other?

Comment (yihezkel) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java:12]: I guess they should then just use the default constructor followed by the appropriate setter, NBD.

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    public ValidationImplications getValidationPolicyType() {
+        return validationPolicyType;
+    }
+
+    public void setValidationPolicyType(ValidationImplications validationPolicyType) {
+        this.validationPolicyType = validationPolicyType;
+    }
+
+    public enum ValidationOptions {
+        DoNotValidate("DoNotValidate"),
```
Comment (yihezkel) [inline ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java:39]: Java standard is enums in ALL CAPS. The value in the parens allow using the case our service expects.

**Review by yihezkel — CHANGES_REQUESTED** [review]
2 minor changes requested (1 change x 2 enums)

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java (hunk 2 above)
Comment (AsafMah) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java]: Weird that IDEA ignores that by default, thanks!

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java (hunk 1 above)
Comment (AsafMah) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/ValidationPolicy.java:13]: So I'm resolving.

**Review by yihezkel — APPROVED** [review]

**Pull request merged**